src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private final double radius;
+    private final ColorSupplier color;
+
+    public Circle(ColorSupplier color, double radius) {
+        this.color = color;
+        this.radius = radius;
+    }
+
+    @Override
+    public double calculateArea() {
+        return Math.PI * radius * radius;
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: circle, area: "
+                + calculateArea() + "sq. units, radius: "
+                + radius + " units, "
+                + "color: " + color);
```

Comment:
Use name() for getting String representation of enum constants. The 'color' field should be of type Color (enum), and you should call color.name() to get its string representation.